tests/tests/test_mtls.py

```diff
@@ -191,6 +259,40 @@ def test_mtls_enterprise(self, setup_ent_mtls, algorithm):
         out = device.run("mender -show-artifact").strip()
         assert out == "mtls-artifact"
 
+    @MenderTesting.fast
+    @pytest.mark.parametrize("algorithm", ["rsa"])
+    def test_mtls_enterprise_hsm(self, setup_ent_mtls, algorithm):
+        # set up the mTLS test environment and return the device
+        device = self.common_test_mtls_enterprise(setup_ent_mtls, algorithm, True)
+
+        # prepare a test artifact
+        with tempfile.NamedTemporaryFile() as tf:
+            artifact = make_script_artifact(
+                "mtls-artifact", conftest.machine_name, tf.name
+            )
+            deploy.upload_image(artifact)
+
+        # deploy the update to the device
+        devices = list(
+            set([device["id"] for device in devauth.get_devices_status("accepted")])
+        )
+        assert len(devices) == 1
+        deployment_id = deploy.trigger_deployment(
+            "mtls-test", artifact_name="mtls-artifact", devices=devices,
+        )
+
+        # now just wait for the update to succeed
+        deploy.check_expected_statistics(deployment_id, "success", 1)
+        deploy.check_expected_status("finished", deployment_id)
+
+        # verify the update was actually installed on the device
+        out = device.run(
+            "export SOFTHSM2_CONF=/softhsm/softhsm2.conf; mender -show-artifact"
+        ).strip()
+        assert out == "mtls-artifact"
+
+        self.hsm_cleanup(device)
+
```

Comment:
Is it enough with checking that the Artifact was installed? Probably yes, but I was expecting to check something from the Mender client logs that would ensure we are loading the keys actually from the HSM.

Reply:
you are reading my mind, thats the second followup I mentioned during su. I will take care of it.

Reply:
Nice. So the idea could be:

```
output = device.run("journalctl -u %s" % device.get_client_service_name())
assert "something" in output
```

Reply:
done. now this I mentioned during su as a follow up. the story: I had put in for this very purpose a like of logs in the mender PR from previous week, but there is a misspell there, it says:
```
time="2020-09-04T09:51:32Z" level=info msg="laoded private key: '&{0xc43560 0xc000010a30}' from 'pkcs11'."
```
so I will file a PR and then correct also here. right now I am checking for "ded private key: " to be present in the logs.

and the test will not pass if the key cannot be loaded -- thats the whole idea of mtls test and thats why i chose to add HSM private key loading testing in here.

Reply:
Got it. Thanks!